BaseListener/Infrastructure/HttpApiContext.cs

```diff
@@ -0,0 +1,61 @@
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.AspNetCore.WebUtilities;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BaseListener.Infrastructure
+{
+    public class HttpApiContext : IHttpApiContext
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public HttpApiContext(IHttpClientFactory httpClientFactory)
+        {
+            this._httpClientFactory = httpClientFactory;
+        }
+
+        private HttpClient Setup(APIGatewayProxyRequest apiGatewayProxyRequest, string method)
+        {
+            string authorization;
+
+            apiGatewayProxyRequest.Headers.TryGetValue("Authorization", out authorization);
+
+            var httpClient = _httpClientFactory.CreateClient();
+
+            httpClient.DefaultRequestHeaders.Add("Authorization", authorization);
```

Comment:
Interesting. I don't think ever saw the auth values setup that way.
How does this work? Where does the authorization bearer token come from?

Reply:
Assuming we're passing in the bearer token values via here....

It's not the case on development and staging, but on pretty sure on production for some of the APIs I saw, we have Bearers with different permission levels. It was made so that 1 Bearer token could have the permissions of GET, and another permission would have the permissions of POST. So you need more than 1 Bearer to use the same API.

Will this HttpClient setup work with these constraints?

Reply:
Done